addons/mail/models/mail_activity.py

```diff
@@ -241,6 +242,20 @@ def action_feedback(self, feedback=False):
         message = self.env['mail.message']
         if feedback:
             self.write(dict(feedback=feedback))
+
+        # Search for all attachments linked to the activities we are about to unlink. This way, we
+        # can link them to the message posted and prevent their deletion.
+        attachments = self.env['ir.attachment'].search_read([
+            ('res_model', '=', self._name),
+            ('res_id', 'in', self.ids),
+        ], ['id', 'res_id'])
+
+        # Filling a dict containing the list of attachment per activity.
+        activity_attachments = defaultdict(list)
+        for attachment in attachments:
+            value, key = attachment.values()
```

Comment:
please use proper variable names, so you won't need the comment

Reply:
Also this relies on an implicit ordering the entries in the dict, and nothing guarantees that `'id'` will always be added before `'res_id'`. In fact before Python 3.6 the order was randomized entirely. You should make that explicit.
Maybe you can use `read_group` to do the grouping for you?

Reply:
`read_group(..."ids:array_agg(id)"...` doesn't seem to work in 11.0 and in 13.0 we get explicit error:

https://github.com/odoo/odoo/blob/720a68a275163386e1b49dd03bb59f671f29f55c/odoo/addons/base/models/ir_attachment.py#L391-L392

so explicitly getting `res_id` and `id` seems better

Reply:
I renamed the variables and called res_id & id explicitly

Reply:
:+1: safer now, and more readable :)